website/docs/advanced-concepts/version-control-with-git/README.mdx

```diff
@@ -0,0 +1,102 @@
+
+### Generate a deploy key
+
+1. Open the app you want to connect to git and click on the `Connect GIT` button on the bottom bar.
+2. In the Git connection window, paste your repository’s `SSH` URL and click on `Generate key` to generate an `SSH` key.
+
+
+
+ <VideoEmbed host="youtube" videoId="oFfdIwhSPL8" title="Generating a Deploy key" caption="Generating a Deploy key"/>
+
+Appsmith supports two types of keys -  **ECDSA 256** and **RSA 4096**. Please pick the one that's compatible with your Git hosting service. 
+
+### Adding the deploy key in the Repository
+
+Copy the generated deploy key and add it to your repository. Please note that you have to **enable write access** for pushing the changes to the remote repository from Appsmith.
+
+<Tabs queryString="current-edition">
+<TabItem label="GitHub" value="github">
+
+* Open the settings of the repository and go to deploy keys.
+* Click **Add deploy key**. Add a title for the key and paste the generated key into the key section.
+* Check the **Allow write access** checkbox and add the key.
+
+  <VideoEmbed host="youtube" videoId="4xMHO4G2hEA" title="Adding the deploy key in the Repository | Github" caption="Adding the deploy key in the Repository | Github"/>
+
+</TabItem>
+<TabItem label="Gitlab" value="gitlab">
+
+* Open the Settings> Repository and expand the deploy keys section.
+* Click **Add key** Add the title and paste the generated key in the key section.
+* Check the **Grant write permissions**" to this key and add the key.
+
+ <VideoEmbed host="youtube" videoId="9aaiE6OERW0" title="Adding the deploy key in the Repository | Gitlab" caption="Adding the deploy key in the Repository | Gitlab"/>
+
+</TabItem>
+<TabItem label="Bitbucket" value="bitbucket">
+
+* Open **Personal settings** from the bottom right corner and go to **SSH keys**.
+* Click **Add key**, add the label and paste the generated key in the key section.
+
+ <VideoEmbed host="youtube" videoId="A8ZOvW1CVIk" title="Adding the deploy key in the Repository | Bitbucket" caption="Adding the deploy key in the Repository | Bitbucket"/>
+
+</TabItem>
+</Tabs>
+
+:::info
+**AWS CodeCommit** isn't supported as a source control repository due to their lack of support for deploy keys.
+:::
+
+### User configurations
+
+* In user settings, define the user configurations. By default, the global user configurations are used. Click the **Edit** button or directly open the profile section to change the global configurations. If you want to add a user configuration specific to the current app, you must clear the **use default configuration** checkbox and enter the author details below.
+
+ <VideoEmbed host="youtube" videoId="d5R1MYKtpCM" title="User Settings | Git Sync" caption="User Settings | Git Sync"/>
+
+Finally, click on **connect**, and when the connection is successful, your repository will have a `README.md` file. On the Git window, you’ll move to the **Deploy** section, where you can make an initial commit.
```

Comment:
this should be part of the step ordering right? Also, let's use number ordering everwhere where it is a procedure.  

And its not just `README.md` file that gets added to the repo. A bunch of stuff including two folders and a few JSON files also get added. 

Also unless they make any changes they don't have to initially commit anything.

Reply:
I think when the connection is successful, a README file gets added. An initial commit should be done to update the re[o the current state of the app. Currently, the repo is empty. The initial commit is a change itself as the repo does not have anything. The initial commit adds the app to the repository. If you don't make the initial commit, the repo will be empty even if the connection is successful.

Reply:
Sorry just checked. The initial commit step is automated now. Previously you had to make the initial commit manually.